virttest/qemu_vm.py

```diff
@@ -1001,15 +1000,10 @@ def set_value(opt_string, key, fallback=None):
                     spice_opts.append(opt_string % tmp)
                 elif fallback:
                     spice_opts.append(fallback)
-            s_port = str(utils_misc.find_free_port(*port_range))
             if optget("spice_port") == "generate":
-                if not self.is_alive():
-                    self.spice_options['spice_port'] = s_port
-                    spice_opts.append("port=%s" % s_port)
-                    self.spice_port = s_port
-                else:
-                    self.spice_options['spice_port'] = self.spice_port
-                    spice_opts.append("port=%s" % self.spice_port)
+                s_port = str(utils_misc.find_free_port(*port_range))
```

Comment:
This is where I'd expect the `FIXME: This makes the "needs_restart" to always re-create the machine...` message.